src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -22,574 +35,658 @@
-				BlockPos outputPos = this.getPos().offset(facing, 2).offset(facing.rotateY().getOpposite(), 2).offset(EnumFacing.DOWN, 1);
-				IFluidHandler output = FluidUtil.getFluidHandler(worldObj, outputPos, facing);
-				if(output != null)
-				{
+				FluidStack out = new FluidStack(availableFluid(),
+						Math.min(IPConfig.Extraction.pumpjack_speed, oilAmnt));
+				BlockPos outputPos = this.getPos().offset(facing, 2)
+						.offset(facing.rotateY().getOpposite(), 2)
+						.offset(EnumFacing.DOWN, 1);
+				IFluidHandler output = FluidUtil.getFluidHandler(worldObj,
+						outputPos, facing);
+				if (output != null) {
 					int accepted = output.fill(out, false);
-					if(accepted > 0)
-					{
-						int drained = output.fill(Utils.copyFluidStackWithAmount(out, Math.min(out.amount, accepted), false), true);
+					if (accepted > 0) {
+						int drained = output.fill(
+								Utils.copyFluidStackWithAmount(out,
+										Math.min(out.amount, accepted), false),
+								true);
 						extractOil(drained);
-						out = Utils.copyFluidStackWithAmount(out, out.amount - drained, false);
+						out = Utils.copyFluidStackWithAmount(out,
+								out.amount - drained, false);
 					}
 				}
-				
-				outputPos = this.getPos().offset(facing, 2).offset(facing.rotateY(), 2).offset(EnumFacing.DOWN, 1);
+
+				outputPos = this.getPos().offset(facing, 2)
+						.offset(facing.rotateY(), 2).offset(EnumFacing.DOWN, 1);
 				output = FluidUtil.getFluidHandler(worldObj, outputPos, facing);
-				if(output != null)
-				{
+				if (output != null) {
 					int accepted = output.fill(out, false);
-					if(accepted > 0)
-					{
-						int drained = output.fill(Utils.copyFluidStackWithAmount(out, Math.min(out.amount, accepted), false), true);
+					if (accepted > 0) {
+						int drained = output.fill(
+								Utils.copyFluidStackWithAmount(out,
+										Math.min(out.amount, accepted), false),
+								true);
 						extractOil(drained);
-	
+
 					}
 				}
-							
-				
+
 				activeTicks++;
 			}
 		}
 
-		if(active != wasActive)
-		{
+		if (active != wasActive) {
 			this.markDirty();
 			this.markContainingBlockForUpdate(null);
 		}
-		
+
 		wasActive = active;
 
 	}
 
+	// Checks whether or not this pump has a pipe to bedrock.
+	private boolean hasPipe() {
+		if (!IPConfig.Extraction.require_pipes) {
+			return true;
+		}
+		
+		World world = this.getWorld();
+		BlockPos pos = this.getOrigin();
+		IBlockState state = world.getBlockState(pos);
+		int damage = state.getBlock().damageDropped(state);
+		String unlocalizedName = state.getBlock().getUnlocalizedName();
+		int depthCount = 0;
+		int depthBounds = 256;
+		while (depthCount < depthBounds) {
+			pos = pos.add(0, -1, 0);
+			state = world.getBlockState(pos);
+			unlocalizedName = state.getBlock().getUnlocalizedName();
+			damage = state.getBlock().damageDropped(state);
+			if (unlocalizedName.equals("tile.bedrock")) {
```

Comment:
Is there something fundamentally wrong with the way I check the damage value then? I can probably rework the check to not use the unlocalized name, and I see why that'd be a good idea, but I don't see any inherit evils in the other check I perform.